src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,204 @@
+     * @param input describing task information.
+     * @return Task object if successfully created and added.
+     * @throws DukeException if failed to add task.
+     */
+    public Task addTask(String type, String input) throws DukeException {
+        return this.addTask(type, input, false);
+    }
+
+    /**
+     * Creates and adds a new task to the TaskList object.
+     *
+     * @param type describing task type.
+     * @param input describing task information.
+     * @param done describing status of task.
+     * @return Task object if successfully created and added.
+     * @throws DukeException if failed to add task.
+     */
+    public Task addTask(String type, String input, boolean done) throws DukeException {
+        Task task;
+        switch (type) {
+        case "todo":
+            task = new Todo(input);
+            break;
+        case "deadline":
+            String[] arr = input.split("/by");
+            try {
+                task = new Deadline(arr[0].trim(), arr[1].trim());
+            } catch (IndexOutOfBoundsException e) {
+                throw new DukeException("OOPS!!! I'm sorry, when is the deadline? (/by...)");
+            } catch (DateTimeParseException e) {
+                throw new DukeException("Invalid date-time format! (/by...)");
+            }
+            break;
+        case "event":
+            arr = input.split("/at");
+            try {
+                task = new Event(arr[0].trim(), arr[1].trim());
+            } catch (IndexOutOfBoundsException e) {
+                throw new DukeException("OOPS!!! I'm sorry, when is the event? (/at...)");
+            } catch (DateTimeParseException e) {
+                throw new DukeException("Invalid date-time format! (/at...)");
+            }
+            break;
+        default:
+            throw new DukeException("OOPS!!! I'm sorry, I don't know what that means :<");
+        }
+        if (input.isEmpty()) {
+            throw new DukeException("OOPS!!! I'm sorry, the description cannot be empty :<");
+        }
+        task = done ? task.setDone() : task;
+        this.list.add(task);
+        return task;
+    }
+
+    /**
+     * Returns a TaskList containing TimedTask instances with the input date.
+     *
+     * @param date date of the timed-tasks we wish to retrieve.
+     * @return TaskList object if successfully retrieved.
+     * @throws DukeException if no tasks on the input date.
+     */
+    public TaskList getTimedTasks(LocalDate date) throws DukeException { // input of form "dd-MM-yyyy"
+        try {
+            ArrayList<Task> timedTasks = new ArrayList<>(this.list);
+            timedTasks.removeIf(i -> i instanceof Todo);
+            Task[] timedTasksArr = timedTasks.toArray(new Task[0]);
+            for (Task i : timedTasksArr) {
+                assert i instanceof TimedTask;
+                if (!((TimedTask) i).getDate().equals(date)) {
+                    timedTasks.remove(i);
```

Comment:
Can consider using streams + filter to filter your TaskList to return a TaskList of only timedTask